frontend/src/metabase/parameters/utils/mapping-options.ts

```diff
@@ -154,15 +157,25 @@ export function getParameterMappingOptions(
 
   const { isNative } = Lib.queryDisplayInfo(question.query());
   if (!isNative) {
-    const { query, stageIndex, columns } = getParameterColumns(
+    const { query, columns } = getParameterColumns(
       question,
       parameter ?? undefined,
     );
-    const columnGroups = Lib.groupColumns(columns);
 
-    const options = columnGroups.flatMap(group =>
-      buildStructuredQuerySectionOptions(query, stageIndex, group),
-    );
+    const groups = _.uniq(columns, ({ group }) => group);
+
+    const options = groups.flatMap(({ stageIndex, group }) => {
+      const groupColumns = columns
+        .filter(column => column.group === group)
+        .map(({ column }) => column);
```

Comment:
Nit: I think this code can be arguably be written more clearly by using `_.groupBy`:

```
const groups = _.groupBy(columns, ({ group }) => group);

const options = groups.flatMap((groupColumns) = {
  // ...
});
```

Reply:
Unfortunately underscore complains that iteratee function passed to `groupBy` needs to return number or a string (`group` is an object).